### PR TITLE
Upgrade NodeJS

### LINKED ISSUE #1083: Update node
node 12 is EOL at the end of this month: https://nodejs.org/en/about/releases/

### PR DESCRIPTION
**Pull request by eliotjordan**
Upgrade NodeJS to 16 LTS

Closes #1083

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -84,17 +84,6 @@ jobs:
           at: '~/pulmap'
       # Install Bundler
       - run: gem install bundler -v '~> 2.0'
-      - run:
```
Comment (tpendragon) [inline .circleci/config.yml:87]: We don't need this one anymore?

**Review by tpendragon — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (eliotjordan) [reply .circleci/config.yml:87]: Shouldn't it be fine to just set the node version in the build step?

Comment (eliotjordan) [reply .circleci/config.yml:87]: It's what we do in Pulfalight: https://github.com/pulibrary/pulfalight/blob/main/.circleci/config.yml#L15

Comment (eliotjordan) [reply .circleci/config.yml:87]: Maybe the Pulfalight config is a mistake.

Comment (eliotjordan) [reply .circleci/config.yml:87]: Yep. Still needed. Added back with new node version.

**Review by tpendragon — APPROVED** [review]

**Pull request merged**